src/main.py

```diff
@@ -241,126 +254,139 @@ def get_post_type(post):
+                                           text="The second parameter has to be a positive integer value. Aborting.")
             return
         if offset < 0:
-            context.bot.send_message(chat_id=update.message.chat_id, text="The second parameter has to be a positive integer value. Aborting.")
+            await context.bot.send_message(chat_id=update.message.chat_id,
+                                           text="The second parameter has to be a positive integer value. Aborting.")
             return
 
-    send_subreddit_posts(subreddit, update, context)
+    await send_subreddit_posts(subreddit, update, context)
 
 
-def rr(update, context):
-    reddit = praw.Reddit(client_id=REDDIT_BOT_ID, client_secret=REDDIT_BOT_SECRET, user_agent=REDDIT_USER_AGENT)
-    sub = reddit.random_subreddit(nsfw=False)
+async def rr(update, context):
+    reddit = asyncpraw.Reddit(client_id=REDDIT_BOT_ID, client_secret=REDDIT_BOT_SECRET, user_agent=REDDIT_USER_AGENT)
+    sub = await reddit.random_subreddit(nsfw=False)
     sub_name = sub.display_name
-    context.bot.send_message(chat_id=update.message.chat_id, text="Random subreddit: \"" + sub_name + "\"")
-    send_subreddit_posts(sub_name, update, context)
+    await context.bot.send_message(chat_id=update.message.chat_id, text="Random subreddit: \"" + sub_name + "\"")
+    await send_subreddit_posts(sub_name, update, context)
 
 
-def cat(update, context):
-    context.bot.send_photo(
+async def cat(update, context):
+    await context.bot.send_photo(
         chat_id=update.message.chat_id,
         photo="https://thiscatdoesnotexist.com?time=" + str(time.time()) + str(random.randint(1, 1024))
     )
 
 
-def snack(update, context):
-    snack_data = requests.get("https://thissnackdoesnotexist.com/?time=" + str(time.time()) + str(random.randint(1, 1024)), headers={'User-Agent': 'USER_AGENT_BROWSER'})
+async def snack(update, context):
+    snack_data = requests.get(
+        "https://thissnackdoesnotexist.com/?time=" + str(time.time()) + str(random.randint(1, 1024)),
+        headers={'User-Agent': 'USER_AGENT_BROWSER'})
     if not snack_data.ok:
-        context.bot.send_message(chat_id=update.message.chat_id, text="Something went wrong internally. I am deeply sorry.")
+        await context.bot.send_message(chat_id=update.message.chat_id,
+                                       text="Something went wrong internally. I am deeply sorry.")
         return
 
-    soup = BeautifulSoup(snack_data.text,'html.parser')
+    soup = BeautifulSoup(snack_data.text, 'html.parser')
     text = soup.find('h1').text
     picture_url = soup.find('div').attrs.get('style').split("(", 1)[1].split(")")[0]
-    context.bot.send_photo(
+    await context.bot.send_photo(
         chat_id=update.message.chat_id,
         photo=picture_url,
         caption=text
     )
 
 
-def horse(update, context):
-    context.bot.send_photo(
+async def horse(update: Update, context: ContextTypes.DEFAULT_TYPE):
```

Comment:
Is there a reason why you indicate types in this method definition but not in others?

Reply:
No just forgot to add it to all methods

Reply:
> > You declared all methods as async.
> > After taking a quick peek at the telegram bot api reference, I think async request handling is not enabled by default. Could you point out the benefit of the requested changes or link the relevant sections of the documentation?
> 
> I got an error when not using async and assumed that it has to be done like that for all methods that send anything to the bot (also not all methods are declared as async). I used an example as reference [doc](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/examples/inlinebot.py)

My bad, I referenced the wrong version of the documentation. Version 20 does in fact require a python coroutine to set up a [`CommandHandler](https://docs.python-telegram-bot.org/en/stable/telegram.ext.commandhandler.html#telegram.ext.CommandHandler.params.callback).

Reply:
Please settle for one of the two styles